lib/eslint.js

```diff
@@ -407,6 +407,30 @@ function isDisabledByReportingConfig(reportingConfig, ruleId, location) {
 }
 
 /**
+ * Normalize ECMAScript version from the initial config
+ * @param  {number} ecmaVersion ECMAScript version from the initial config
+ * @param  {boolean} isModule Whether the source type is module or not
+ * @returns {number} normalized ECMAScript version
+ */
+function normalizeEcmaVersion(ecmaVersion, isModule) {
+
+    // need at least ES6 for modules
+    if (isModule && (!ecmaVersion || ecmaVersion < 6)) {
+        ecmaVersion = 6;
```

Comment:
Are you sure? I thought I was just moving the logic [from here](https://github.com/eslint/eslint/pull/6907/files#diff-b6559e6b095a292d244b6b7bae986291L466) into the normalize function